lib/dal/src/attribute/prototype/debug.rs

```diff
@@ -0,0 +1,207 @@
+use crate::{
+    AttributePrototype, AttributePrototypeId, AttributeValue, AttributeValueId, Component,
+    ComponentError, DalContext, Func, FuncError, FuncId, InputSocket, Prop,
+};
+use serde_json::Value;
+use telemetry::prelude::*;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AttributePrototypeDebugView {
+    pub func_id: FuncId,
+    pub func_execution: Option<FuncExecution>,
+    pub prototype_id: AttributePrototypeId,
+    pub func_name: String,
+    pub func_args: HashMap<String, Vec<serde_json::Value>>,
+    pub arg_sources: HashMap<String, Option<String>>,
+    pub all_attribute_values: Vec<AttributeValueId>,
+}
+
+type AttributePrototypeDebugViewResult<T> = Result<T, AttributePrototypeDebugViewError>;
+
+#[remain::sorted]
+#[derive(Error, Debug)]
+pub enum AttributePrototypeDebugViewError {
+    #[error(transparent)]
+    AttributePrototypeArgumentError(#[from] AttributePrototypeArgumentError),
+    #[error(transparent)]
+    AttributePrototypeError(#[from] AttributePrototypeError),
+    #[error(transparent)]
+    AttributeValue(#[from] AttributeValueError),
+    #[error(transparent)]
+    ComponentError(#[from] ComponentError),
+    #[error(transparent)]
+    Func(#[from] FuncError),
+    #[error(transparent)]
+    FuncExecution(#[from] FuncExecutionError),
+    #[error(transparent)]
+    InputSocketError(#[from] InputSocketError),
+    #[error(transparent)]
+    NodeWeightError(#[from] NodeWeightError),
+    #[error(transparent)]
+    OutputSocketError(#[from] OutputSocketError),
+    #[error(transparent)]
+    PropError(#[from] PropError),
+    #[error(transparent)]
+    ValueSourceError(#[from] ValueSourceError),
+    #[error(transparent)]
+    WorkspaceSnapshotError(#[from] WorkspaceSnapshotError),
+}
+impl AttributePrototypeDebugView {
+    #[instrument(level = "debug", skip_all)]
+    pub async fn new_from_attribute_value(
+        ctx: &DalContext,
+        attribute_value_id: AttributeValueId,
+    ) -> AttributePrototypeDebugViewResult<AttributePrototypeDebugView> {
+        let prototype_id = AttributeValue::prototype_id(ctx, attribute_value_id).await?;
+        let destination_component_id =
+            AttributeValue::component_id(ctx, attribute_value_id).await?;
+        let apa_ids = AttributePrototypeArgument::list_ids_for_prototype(ctx, prototype_id).await?;
+        let func_id = AttributePrototype::func_id(ctx, prototype_id).await?;
+        let all_attribute_values =
+            AttributePrototype::attribute_value_ids(ctx, prototype_id).await?;
+        let mut func_binding_args: HashMap<String, Vec<serde_json::Value>> = HashMap::new();
+        let mut arg_sources: HashMap<String, Option<String>> = HashMap::new();
+
+        let func_name = Func::get_by_id(ctx, func_id).await?.name;
+
+        let func_execution =
+            Some(FuncExecution::get_latest_execution_by_func_id(ctx, &func_id).await?);
+        for apa_id in apa_ids {
```

Comment:
nit: feel free to ignore. I am not a fan of abbreviations, but this leans less into convention and more into our codebase style. We generally avoid abbreviations to prevent convention, help new hires and new contributors, and prevent shoring of information behind abbreviated abstractions.

Relevant doc (kinda?): https://rust-lang.github.io/api-guidelines/naming.html